bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcherImpl.java

```diff
@@ -239,22 +263,40 @@ public BookieSocketAddress replaceBookie(int ensembleSize, int writeQuorumSize,
             throws BKNotEnoughBookiesException {
         long startTime = MathUtils.nowInNano();
         BookieSocketAddress addr = existingBookies.get(bookieIdx);
+        Pair<BookieSocketAddress, Boolean> replaceBookieResponse;
         BookieSocketAddress socketAddress;
+        boolean isEnsembleAdheringToPlacementPolicy = false;
         try {
             // we exclude the quarantined bookies also first
-            Set<BookieSocketAddress> existingAndQuarantinedBookies = new HashSet<BookieSocketAddress>(existingBookies);
-            existingAndQuarantinedBookies.addAll(quarantinedBookies.asMap().keySet());
-            socketAddress = placementPolicy.replaceBookie(
+            Set<BookieSocketAddress> excludeBookiesAndQuarantinedBookies = new HashSet<BookieSocketAddress>(
```

Comment:
you mean 'exclude[d]' bookies?

Reply:
followed existing terminology - 'Set<BookieSocketAddress> excludeBookies', anyhow dont have strong opinion, will change